modules/createRouterHistory.js

```diff
@@ -4,9 +4,43 @@ const canUseDOM = !!(
   typeof window !== 'undefined' && window.document && window.document.createElement
 )
 
+const HISTORY_METHODS = [
+  'listenBefore',
+  'listen',
+  'transitionTo',
+  'push',
+  'replace',
+  'go',
+  'goBack',
+  'goForward',
+  'createKey',
+  'createPath',
+  'createHref',
+  'createLocation'
+]
```

Comment:
Really don't like this. I get that we're tied to the API internally anyways, but this seems like an antipattern to me. I would kill for a `method_missing` equivalent in JS, but that's not going to happen until Proxy is more prevalent.